e2e/test/scenarios/embedding/embedding-reproductions.cy.spec.js

```diff
@@ -1111,6 +1111,71 @@ H.describeEE("issue 8490", () => {
     });
   });
 
+  it("static embeddings with `#locale` should show translate the loading message (metabase#50182)", () => {
+    cy.intercept(
+      {
+        method: "GET",
+        url: "/api/embed/dashboard/*",
+        middleware: true,
+      },
+      req => {
+        req.on("response", res => {
+          const MINUTE = 60 * 1000;
+          res.setDelay(MINUTE);
```

Comment:
When testing in Cypress, the locale is often loaded after the dashboard and the question, so users won't even see the loading indicator. I've tried using `cy.clock`, but it didn't work.

I also try to wait the dashboard and the question request after locale request has finished, but I couldn't managed to pull that off, so one way to test this is to have a separate test that mock the dashboard, and the question request and assert for only the loading indicator.

Reply:
I think I found a way to re-use the normal tests, this is just a test that does it for the dashboard.
It basically makes the request await a promise, and we can resolve the promise after we found the Loading text, will this work?

```
diff --git a/e2e/test/scenarios/embedding/embedding-reproductions.cy.spec.js b/e2e/test/scenarios/embedding/embedding-reproductions.cy.spec.js
index a04fca3dd6a..ee0c225e674 100644
--- a/e2e/test/scenarios/embedding/embedding-reproductions.cy.spec.js
+++ b/e2e/test/scenarios/embedding/embedding-reproductions.cy.spec.js
@@ -1111,35 +1111,36 @@ H.describeEE("issue 8490", () => {
     });
   });
 
-  it("static embeddings with `#locale` should show translate the loading message (metabase#50182)", () => {
+  it.only("static embeddings with `#locale` should show translate the loading message (metabase#50182)", () => {
+    let resolveDelayDashboardRequestPromise;
+    const delayDashboardRequestPromise = new Promise(resolve => {
+      resolveDelayDashboardRequestPromise = resolve;
+    });
+
     cy.intercept(
       {
         method: "GET",
         url: "/api/embed/dashboard/*",
         middleware: true,
       },
-      req => {
-        req.on("response", res => {
-          const MINUTE = 60 * 1000;
-          res.setDelay(MINUTE);
-        });
+      async req => {
+        await delayDashboardRequestPromise;
+        req.continue();
       },
     ).as("dashboardRequest");
-    cy.intercept(
-      {
-        method: "GET",
-        url: "/api/embed/card/*",
-        middleware: true,
-      },
-      req => {
-        req.on("response", res => {
-          const MINUTE = 60 * 1000;
-          res.setDelay(MINUTE);
-        });
-      },
-    ).as("questionRequest");
 
-    cy.log("test a static embedded dashboard");
+    // cy.intercept(
+    //   {
+    //     method: "GET",
+    //     url: "/api/embed/card/*",
+    //     middleware: true,
+    //   },
+    //   async req => {
+    //     await delayDashboardRequestPromise;
+    //     req.continue();
+    //   },
+    // ).as("questionRequest");
+
     cy.get("@dashboardId").then(dashboardId => {
       H.visitEmbeddedPage(
         {
@@ -1155,25 +1156,12 @@ H.describeEE("issue 8490", () => {
     });
 
     // Loading...
-    cy.findByTestId("embed-frame").findByText("로딩...").should("be.visible");
-
-    cy.log("test a static embedded question");
-    cy.get("@lineChartQuestionId").then(lineChartQuestionId => {
-      H.visitEmbeddedPage(
-        {
-          resource: { question: lineChartQuestionId },
-          params: {},
-        },
-        {
-          additionalHashOptions: {
-            locale: "ko",
-          },
-        },
-      );
-    });
-
-    // Loading...
-    cy.findByTestId("embed-frame").findByText("로딩...").should("be.visible");
+    cy.findByTestId("embed-frame")
+      .findByText("로딩...")
+      .should("be.visible")
+      .then(() => {
+        resolveDelayDashboardRequestPromise();
+      });
   });
 
   it("static embeddings should respect `#locale` hash parameter (metabase#8490, metabase#50182)", () => {
@@ -1211,7 +1199,7 @@ H.describeEE("issue 8490", () => {
         // N/A
         cy.findByText("해당 없음").should("be.visible");
         // (No data)
-        cy.findByText("(데이터 없음)").should("be.visible");
+        cy.findByText("(데이터 없���)").should("be.visible");
       });
 
       cy.log("assert the pie chart");

```

Reply:
wall! that's great, I wasn't thinking about using a promise. This looks `Promise`ing

Reply:
@npretto Fixed in 54aad3f65984b3449b139742a4eac0e627b1394e, I clean up the promise thing by using `defer` + [returning the promise in `intercept`](https://docs.cypress.io/api/commands/intercept#Returning-a-Promise-2).

Reply:
Consider the PR re-approved from my side 🚀